wordpress-mobile/WordPress-Android
# [Voice to Content] Support device rotation while recording

Fixes https://github.com/Automattic/wordpress-mobile/issues/77

This PR adds support for device rotation 
- Add `fun isRecording()` and `fun isPaused() to `IAudioRecorder` interface to allow checking if the audio recorder is currently recording or paused.
- Update RecordingUseCase to allow access to the values mentioned above
- Update VoiceToContentViewModel to receive the flow and add to the UI state for use in the VoiceToContentScreen
- Update VoiceToContentScreen and MicToStopIcon to use the isRecording value to drive visual components
- Update relevant classes to implement the new methods 
- Replace log.util with AppLog statements in AudioRecorder 
- Only endRecordingSession is one is in session

This PR is the companion app for https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3035. See merge notes.

This branch build upon issue/v2c-show-elapsed-time
> [!IMPORTANT]
> **Merge Instructions**
> - Make sure #20983, #20984, and #20992 have been merged to trunk 
> - Make sure https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3035 has been merged to trunk
> - Grab the FluxC hash from the CI merge to trunk
> - Update the fluxC hash in build.gradle & commit
> - Remove the Not Ready for Merge Label
> - Merge as normal

> [!NOTE]
> The following have not yet been implemented:
>  - Overall polishing of the UI
> - Unit tests

-----

## To Test:
- Install and launch the app
- Login with an account that has access to AI credits (Any a8c P2 will have this - other free sites have 20 requests)
- Navigate to Me > Debug Settings and enable the voice_to_content flag (restart the app)
- Navigate to My Site and tap the FAB
- Select the Post with Audio option
- Start the recording session by tapping on the mic icon
- Record a blog post - a few words, maybe include a list, try it out
- Rotate the device
- ✅ Verify the recording view remains on the screen and you are still able to record
- Continue recording some audio
- Tap the done icon and wait for the Edit Post view
- ✅ Verify content reflects the audio content from before and after the device rotation
Note: There is a known issue that sometimes all the content shows in the title instead of the body. This is a GB transformation issue that will be addressed separately.

-----

## Regression Notes

1. Potential unintended areas of impact
The app crashes or doesn't pick up recording where it was left off.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A
-----

## PR Submission Checklist:

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): N/A
